docs/contributing/999-rfcs/2024-02-15-resource-pinning.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Resource Pinning
+description: Protection of resources from accidental deletion/moves
+---
+
+# Resource Pinning
+
+- **Author(s):**: @MarkMcCulloh
+- **Submission Date**: 2024-02-15
+- **Stage**: Proposal
+- **Stage Date**: 2024-02-15
+
+Introduces `pin` as a mechanism to protect resources from accidental deletion/moves in the logical application tree.
+
+## User Journey
+
+Sally Stateful creates a wing file with a bucket
+
+```wing
+// main.w
+bring cloud;
+
+new cloud.Bucket() as "CoolBucket";
+```
+
+because they want to make sure the bucket doesn't get deleted by accident, they pin it
+
+```wing
+// main.w
+bring cloud;
+
+pin(new cloud.Bucket() as "CoolBucket");
```

Comment:
kind of feels like this should be `pinned()` because it returns the object, right?

Reply:
It does return the object, but I think `pin()` is fine. Like `unsafeCast()`. I'm open to either since I can see why `pinned()` would make sense but chose the shorter verb option.

Reply:
I am wondering if the pinning should be on the id:

```js
new cloud.Bucket() as pinned("CoolBucket");
```

Reply:
This was my first thought too, but symantically it's a little confusing (is pinned a function called on a string?). I also considered `as pinned "CoolBucket"`. Also I don't know where https://github.com/winglang/wing/issues/4604 is going to end up, which could effect this as well.

Reply:
Updated to use .pinned syntax